src/MgvOfferTaking.sol

```diff
@@ -306,146 +306,97 @@ abstract contract MgvOfferTaking is MgvHasOffers {
-        mor.totalGot = 0;
-        mor.totalGave = 0;
-
-        /* Initialize single order struct. */
-        sor.offerId = targets[i][0];
-        OfferData storage offerData = pair.offerData[sor.offerId];
-        sor.offer = offerData.offer;
-        sor.offerDetail = offerData.detail;
-
-        /* If we removed the `isLive` conditional, a single expired or nonexistent offer in `targets` would revert the entire transaction (by the division by `offer.gives` below since `offer.gives` would be 0). We also check that `gasreq` is not worse than specified. A taker who does not care about `gasreq` can specify any amount larger than $2^{24}-1$. A mismatched price will be detected by `execute`. */
-        if (!sor.offer.isLive() || sor.offerDetail.gasreq() > targets[i][3]) {
-          /* We move on to the next offer in the array. */
-          continue;
-        } else {
-          {
-            Tick tick = Tick.wrap(int(targets[i][1]));
-            require(TickLib.inRange(tick), "mgv/snipes/tick/outOfRange");
-            mor.maxTick = tick;
-          }
-          {
-            uint fillVolume = targets[i][2];
-            require(uint96(fillVolume) == fillVolume, "mgv/snipes/volume/96bits");
-            mor.fillVolume = fillVolume;
-          }
+      emit OrderStart();
 
-          /* We start be enabling the reentrancy lock for this (`outbound_tkn`,`inbound_tkn`) pair. */
-          sor.local = sor.local.lock(true);
-          pair.local = sor.local;
+      /* We start be enabling the reentrancy lock for this (`outbound_tkn`,`inbound_tkn`) pair. */
+      sor.local = sor.local.lock(true);
+      pair.local = sor.local;
 
-          /* `execute` will adjust `sor.wants`,`sor.gives`, and may attempt to execute the offer if its price is low enough. It is crucial that an error due to `taker` triggers a revert. That way [`mgvData`](#MgvOfferTaking/statusCodes) not in `["mgv/tradeSuccess","mgv/notExecuted"]` means the failure is the maker's fault. */
-          /* Post-execution, `sor.wants`/`sor.gives` reflect how much was sent/taken by the offer. */
-          (uint gasused, bytes32 makerData, bytes32 mgvData) = execute(pair, mor, sor);
+      {
+        /* `execute` will adjust `sor.wants`,`sor.gives`, and may attempt to execute the offer if its price is low enough. It is crucial that an error due to `taker` triggers a revert. That way [`mgvData`](#MgvOfferTaking/statusCodes) not in `["mgv/tradeSuccess","mgv/notExecuted"]` means the failure is the maker's fault. */
+        /* Post-execution, `sor.wants`/`sor.gives` reflect how much was sent/taken by the offer. */
+        (uint gasused, bytes32 makerData, bytes32 mgvData) = execute(pair, mor, sor);
 
-          if (mgvData == "mgv/tradeSuccess") {
-            successCount += 1;
-          }
+        require(mgvData != "mgv/tradeSuccess", "mgv/clean/offerDidNotFail");
```

Comment:
This looks correct - however, would it be safer to explicitly check the allowed failure values? 
`require(mgvData == "mgv/makerRevert" || mgvData == "mgv/makerTransferFail" || mgvData == "mgv/makerReceiveFail", "mgv/clean/offerDidNotFail");

Reply:
The intent was to be passively future-proof against additional errors that would be added later. I think that still applies so I would tend towards keeping it as is.

Reply:
Hmm, the point here is to revert explicitly if the offer succeeds, because then it shouldn't be cleaned.
But maybe that isn't clear?